tests/test_cpio.py

```diff
@@ -115,3 +115,43 @@ def test_xz(self):
             raise unittest.SkipTest("lzma package or xz tool not available")
         print 'Running test for XZ'
         self.doArchive('archive.cpio.xz', 'xz')
+
+    # CpioFileCompat testing
+
+    def archiveExtractCompat(self, fn, comp):
+        arc = CpioFileCompat(fn, mode="r", compression={"": CPIO_PLAIN,
```

Comment:
As an enhancement it'd be nice to get a context manager to close the resource when reaching the end of the scope instead of forcing users to close them every time.

Reply:
100% agree.  This feature has been added to `tarfile.py` in python3, and we should be able to get it, likely together with other goodies, but we would first have to sync with the 2.7 version, and we don't even have a clue which version was used as a base for this work :/

Reply:
Python 2.2

Reply:
OK, but that will likely be for a separate PR, let's not mix this with the tests stuff :)

Reply:
I think we want to stay as close to upstream as possible, and get the <2.7-isms removed from cpython's `tarfile` first.  We'll then get them into `cpiofile` by merging newer upstream versions.